rails/app/views/dashboard/places/_form.html.erb

```diff
@@ -34,11 +34,57 @@
     <%= f.collection_check_boxes :story_ids, current_community.stories, :id, :title %>
   </div>
 
-  <%= f.label :lat %>
-  <%= f.text_field :lat %>
+  <div class="two-columns">
+    <div class="two-columns-left column-fixed-400">
+      <%= f.label :lat %>
+      <%= f.text_field :lat %>
 
-  <%= f.label :long %>
-  <%= f.text_field :long %>
+      <%= f.label :long %>
+      <%= f.text_field :long %>
+    </div>
+    <div class="two-columns-right">
+      <div id="placeMap"></div>
+    </div>
+  </div>
 
   <%= f.submit %>
 <% end %>
+
+<script>
+const placeForm = document.querySelector("#place_form");
+const placeLatInput = placeForm.querySelector("#place_lat");
+const placeLongInput = placeForm.querySelector("#place_long");
+const coordinates = [placeLongInput.value, placeLatInput.value]
+const placeMap = new mapboxgl.Map({
+  accessToken: "<%= current_community.theme.mapbox_token %>",
+  container: "placeMap", // container ID
+  center: coordinates, // starting position [lng, lat]
+  hash: false,
+  zoom: 3, // starting zoom
```

Comment:
Let's set this to `5`. `3` is probably a little zoomed out for most use cases; with `5` we can still make out a good amount of detail in the default offline vector tiles, but is more focused on a region.

Reply:
ok no problem, but I think if we make this 5 we should maybe make the initial position something that isn't null island 0, 0 because all you can see is ocean and it's not super obvious you can interact with it. Here's a screenshot

<img width="786" alt="image" src="https://github.com/Terrastories/terrastories/assets/217050/1865d93a-6453-45aa-bcfe-92e05fe7c347">

Reply:
Ah, good point. This actually makes me think we're better off using the zoom level and center lat/longs from the community map settings. Null island is not a useful reference for anyone, and users will have to navigate away from it to their area of interest every time.

I'd say let's ship this with zoom level 7 and have that be another ticket (unless you think it's doable with minimal work having just dug into the code for both of these?)

Reply:
It was pretty easy to use the community theme's zoom and lat/long. I implemented it in https://github.com/Terrastories/terrastories/pull/922/commits/b8d8f1897aedd30a99b1be4f7c9c94f4c06809ae

Reply:
Awesome! We just need one small fix, which is to implement the change for the edit form - the Zoom value there is showing as null.

![image](https://github.com/Terrastories/terrastories/assets/31662219/7a033429-e9d9-460a-b666-583bc6b4527b)

Reply:
Oh great catch, I was debating whether to use an instance variable or make it more consistent with the other values pulled off the theme there, but forgot for edit instead of new we wouldn't have access to the instance variable. I pushed a fix for it in https://github.com/Terrastories/terrastories/pull/922/commits/55f88167764a49ffee4c6dad7c498284e293e4c7

I also remembered I should probably handle the case where Theme is missing or has null values for lat/long/zoom. It looks like we should always have a theme thanks to this after_create hook https://github.com/Terrastories/terrastories/blob/ca68805a53313313d8c217768a2973982b430c83/rails/app/models/community.rb#L12, but it does look like it's possible for the other values to be null, so I added fallbacks to 0, 0 with zoom 5. Hopefully most users set up their theme before adding places so this won't be much of an issue.